drake/systems/DrakeSystem.h

```diff
@@ -0,0 +1,124 @@
+#ifndef __DrakeSystem_H__
+#define __DrakeSystem_H__
+
+#include <string>
+#include <memory>
+#include <stdexcept>
+#include <Eigen/Dense>
+#include "CoordinateFrame.h"
+
+#undef DLLEXPORT
+#if defined(WIN32) || defined(WIN64)
+#if defined(drakeSystem_EXPORTS)
+#define DLLEXPORT __declspec( dllexport )
+#else
+#define DLLEXPORT __declspec( dllimport )
+#endif
+#else
+#define DLLEXPORT
+#endif
+
+
+/// A dynamical system authored in Drake registers it's dynamics as well as information
+/// about it's coordinate frames.
+
+class DLLEXPORT DrakeSystem : public std::enable_shared_from_this<DrakeSystem> {
+public:
+
+  typedef Eigen::Matrix<double,Eigen::Dynamic,1> VectorXs;
```

Comment:
I imagine this is going to depend on a template parameter? Otherwise just use VectorXd.